-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b50767/34416/summary.html
**COMMIT**: 2c532f00960608e4db15fafc429611ad83bbe61a
**CMSSW**: CMSSW_13_3_ROOT6_X_2023-08-21-2300/el8_amd64_gcc11
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/42628/34416/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_ROOT6_X_2023-08-21-2300/src/DQM/TrackingMonitor/src/V0Monitor.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_ROOT6_X_2023-08-21-2300/src/DQM/TrackingMonitor/src/VertexMonitor.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_ROOT6_X_2023-08-21-2300/src/DQM/TrackingMonitor/src/dEdxAnalyzer.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_ROOT6_X_2023-08-21-2300/src/DQM/TrackingMonitor/src/dEdxHitAnalyzer.cc
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_ROOT6_X_2023-08-21-2300/src/DQM/TrackingMonitor/src/TrackAnalyzer.cc: In member function 'void tadqm::TrackAnalyzer::bookHistosForHitProperties(dqm::legacy::DQMStore::IBooker&)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_ROOT6_X_2023-08-21-2300/src/DQM/TrackingMonitor/src/TrackAnalyzer.cc:718:33: error: invalid application of 'sizeof' to incomplete type 'const string []' {aka 'const std::__cxx11::basic_string<char> []'}
  718 |     size_t StopReasonNameSize = sizeof(StopReasonName::StopReasonName) / sizeof(std::string);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_ROOT6_X_2023-08-21-2300/src/DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc: In member function 'void TrackBuildingAnalyzer::initHisto(dqm::legacy::DQMStore::IBooker&, const edm::ParameterSet&)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_13_3_ROOT6_X_2023-08-21-2300/src/DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc:332:33: error: invalid application of 'sizeof' to incomplete type 'const string []' {aka 'const std::__cxx11::basic_string<char> []'}
  332 |     size_t StopReasonNameSize = sizeof(StopReasonName::StopReasonName) / sizeof(std::string);
</pre>

--- a/DQM/TrackingMonitor/src/TrackAnalyzer.cc
+++ b/DQM/TrackingMonitor/src/TrackAnalyzer.cc
@@ -715,7 +715,7 @@ void TrackAnalyzer::bookHistosForHitProperties(DQMStore::IBooker& ibooker) {
       oriAlgo->setBinLabel(ibin + 1, reco::TrackBase::algoNames[ibin]);
     }
 
-    size_t StopReasonNameSize = sizeof(StopReasonName::StopReasonName) / sizeof(std::string);
+    size_t StopReasonNameSize = static_cast<size_t>(StopReason::SIZE);
     histname = "stoppingSource_";
     stoppingSource = ibooker.book1D(
         histname + CategoryName, histname + CategoryName, StopReasonNameSize, 0., double(StopReasonNameSize));

--- a/DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc
+++ b/DQM/TrackingMonitor/src/TrackBuildingAnalyzer.cc
@@ -329,7 +329,7 @@ void TrackBuildingAnalyzer::initHisto(DQMStore::IBooker& ibooker, const edm::Par
 
   if (doAllTCPlots || doStopSource) {
     // DataFormats/TrackReco/interface/TrajectoryStopReasons.h
-    size_t StopReasonNameSize = sizeof(StopReasonName::StopReasonName) / sizeof(std::string);
+    size_t StopReasonNameSize = static_cast<size_t>(StopReason::SIZE);
 
     histname = "StoppingSource_" + seedProducer.label() + "_";
     stoppingSource = ibooker.book1D(

--- a/DataFormats/TrackCandidate/interface/TrajectoryStopReasons.h
+++ b/DataFormats/TrackCandidate/interface/TrajectoryStopReasons.h
@@ -22,21 +22,7 @@ enum class StopReason {
 
 // to be kept in synch w/ the above enum ;)
 namespace StopReasonName {
-  static const std::string StopReasonName[] = {
-      "UNINITIALIZED",                 //  0
-      "MAX_HITS",                      //  1
-      "MAX_LOST_HITS",                 //  2
-      "MAX_CONSECUTIVE_LOST_HITS",     //  3
-      "LOST_HIT_FRACTION",             //  4
-      "MIN_PT",                        //  5
-      "CHARGE_SIGNIFICANCE",           //  6
-      "LOOPER",                        //  7
-      "MAX_CCC_LOST_HITS",             //  8
-      "NO_SEGMENTS_FOR_VALID_LAYERS",  //  9
-      "SEED_EXTENSION",                // 10
-      "NOT_STOPPED"                    // 11 (be careful, NOT_STOPPED needs to be the last,
-                                       //     its index differs from the enumeration value)
-  };
+  extern const std::string StopReasonName[];
 };
 
 #endif

--- a/DataFormats/TrackCandidate/src/TrajectoryStopReasons.cc
+++ b/DataFormats/TrackCandidate/src/TrajectoryStopReasons.cc
@@ -1,5 +1,21 @@
 #include "DataFormats/TrackCandidate/interface/TrajectoryStopReasons.h"
 
+const std::string StopReasonName::StopReasonName[] = {
+    "UNINITIALIZED",                 //  0
+    "MAX_HITS",                      //  1
+    "MAX_LOST_HITS",                 //  2
+    "MAX_CONSECUTIVE_LOST_HITS",     //  3
+    "LOST_HIT_FRACTION",             //  4
+    "MIN_PT",                        //  5
+    "CHARGE_SIGNIFICANCE",           //  6
+    "LOOPER",                        //  7
+    "MAX_CCC_LOST_HITS",             //  8
+    "NO_SEGMENTS_FOR_VALID_LAYERS",  //  9
+    "SEED_EXTENSION",                // 10
+    "NOT_STOPPED"                    // 11 (be careful, NOT_STOPPED needs to be the last,
+                                     //     its index differs from the enumeration value)
+};
+
 static_assert(sizeof(StopReasonName::StopReasonName) / sizeof(std::string) ==
                   static_cast<unsigned int>(StopReason::SIZE),
               "StopReason enum and StopReasonName are out of synch");

--- a/DataFormats/TrackReco/interface/SeedStopReason.h
+++ b/DataFormats/TrackReco/interface/SeedStopReason.h
@@ -15,15 +15,7 @@ enum class SeedStopReason {
 };
 
 namespace SeedStopReasonName {
-  static const std::string SeedStopReasonName[] = {
-      "UNINITIALIZED",        // 0
-      "NOT_STOPPED",          // 1
-      "SEED_CLEANING",        // 2
-      "NO_TRAJECTORY",        // 3
-      "SEED_REGION_REBUILD",  // 4
-      "FINAL_CLEAN",          // 5
-      "SMOOTHING_FAILED"      // 6
-  };
+  extern const std::string SeedStopReasonName[];
 }
 
 #endif

--- a/DataFormats/TrackReco/src/SeedStopReason.cc
+++ b/DataFormats/TrackReco/src/SeedStopReason.cc
@@ -1,5 +1,15 @@
 #include "DataFormats/TrackReco/interface/SeedStopReason.h"
 
+const std::string SeedStopReasonName::SeedStopReasonName[] = {
+    "UNINITIALIZED",        // 0
+    "NOT_STOPPED",          // 1
+    "SEED_CLEANING",        // 2
+    "NO_TRAJECTORY",        // 3
+    "SEED_REGION_REBUILD",  // 4
+    "FINAL_CLEAN",          // 5
+    "SMOOTHING_FAILED"      // 6
+};
+
 static_assert(sizeof(SeedStopReasonName::SeedStopReasonName) / sizeof(std::string) ==
                   static_cast<unsigned int>(SeedStopReason::SIZE),
               "SeedStopReason enum and SeedStopReasonName are out of synch");

